pallets/ajuna-board/src/lib.rs

```diff
@@ -224,6 +238,12 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::finish_game())]
 		pub fn finish_game(origin: OriginFor<T>, board_id: T::BoardId) -> DispatchResult {
 			// TODO if this is L2 do we really need to check the origin?
+			//
+			// There is no check for the "origin", so is anyone able to finish any game?
+			// What if the game is still running (no winner)?
+			// Maybe we should allow the game's creator and its players to finish it.
+			// Or finish it right after "declare_winner()" is called, as we already deposit an
+			// "Event::GameFinished" there.
```

Comment:
Yeah we need to keep track of which origin created games so only they can finish games (related to the comment above).

Just sharing more context about this pallet - all the extrinsics in this pallet should be signed and checked by TEE enclave key, which isn't supported yet. And this pallet belongs to L2 runtime under TEE.

Reply:
Just to add, this pallet is executed in a Trusted Execution Environment (TEE) and probably should be pulled out into its own repo to make this clear as the pallet and its environment aren't strictly a substrate runtime.  In other words there are small workarounds for this sidechain.  Having said all of that your point is valid and we intend to, as @cowboy-bebug has pointed out, to put in a guard for this at a later date.